tests/utils/test_decorators.py

```diff
@@ -115,6 +115,130 @@ def my_kernel(x, backend="cudnn"):
     assert my_kernel.is_compute_capability_supported(70) is False  # neither has it
 
 
+def test_backend_requirement_empty_backends_with_common_check_cc():
+    """Test backend_requirement with empty backend_checks but common_check with compute capability."""
+
+    # Made up compute capability
+    @supported_compute_capability([42])
+    def _common_check(x):
+        # Common check with compute capability restrictions
+        return x.shape[0] <= 1024
+
+    @backend_requirement(
+        {},  # Empty backend_checks
+        common_check=_common_check,
+    )
+    def unsupported_kernel(x):
+        return x * 2
+
+    # Check methods
+    assert hasattr(unsupported_kernel, "is_backend_supported")
+    assert hasattr(unsupported_kernel, "is_compute_capability_supported")
+
+    # Check compute capability support (only common_check)
+    assert unsupported_kernel.is_compute_capability_supported(42) is True
+    assert unsupported_kernel.is_compute_capability_supported(75) is False
+    # Undefined behaviour: is_backend_supported with no backend choices
+    with pytest.raises(
+        ValueError,
+        match="Invalid is_backend_supported call: no backend choices for unsupported_kernel",
+    ):
+        unsupported_kernel.is_backend_supported(None, 42)
+        unsupported_kernel.is_backend_supported(None, 75)
+
```

Comment:
[nit] maybe also check the output of this call for a string. I see you did this later also for "random_backend"